### ISSUE TITLE
Question: Why extending `@typescript-eslint/recommended` automatically adds tsx and ts extensions?

### ISSUE THREAD

**Issue by fregante** [typescript-eslint/typescript-eslint#3824]
Sorry, this is just a question. As far as I know eslint only runs on JS files and I have to use `eslint --ext ts,tsx` to run it on TypeScript files. There's no way around it… as far as I know. This is what your own docs say.

But actually I run eslint with just `@typescript-eslint/recommended` and eslint automatically picks up ts and tsx files. Why and most importantly how?!

Repro:

```sh
  npm init -y
  npm install eslint  @typescript-eslint/parser @typescript-eslint/eslint-plugin
  echo '// @ts-ignore' > index.ts
  echo '{}' > tsconfig.json
  echo '{"extends":["plugin:@typescript-eslint/recommended"]}' > .eslintrc
  npx eslint .
```

```
./index.ts
  1:1  error  Do not use "@ts-ignore" because it alters compilation errors  @typescript-eslint/ban-ts-comment

✖ 1 problem (1 error, 0 warnings)
```

🤔 

We were wondering in https://github.com/xojs/eslint-config-xo-typescript/issues/39

**Comment by bradzacher** [comment]
It's likely because of our `eslint-recommended` config which explicitly disables/enables some rules on `.ts`/`.tsx` files:
https://github.com/typescript-eslint/typescript-eslint/blob/26de6459a38389fff86351c1aa0881b11dc2a76a/packages/eslint-plugin/src/configs/eslint-recommended.ts#L9

This causes ESLint to recognise we want to lint `.ts`/`.tsx` files, and thus it automatically includes them when looking for files.
Then because the recommended config contains no restrictions - it just (correctly) applies the rules to the files.

**Comment by fregante** [comment]
I have no idea how I missed that. I was just looking at this file https://github.com/typescript-eslint/typescript-eslint/blob/26de6459a38389fff86351c1aa0881b11dc2a76a/packages/eslint-plugin/src/configs/recommended.ts

Thank you for the quick answer!
